components/camel-stomp/src/main/java/org/apache/camel/component/stomp/StompComponent.java

```diff
@@ -55,6 +67,7 @@ protected Endpoint createEndpoint(String uri, String remaining, Map<String, Obje
         if (config.getSslContextParameters() == null) {
             config.setSslContextParameters(retrieveGlobalSslContextParameters());
         }
+        endpoint.setHeaderFilterStrategy(getHeaderFilterStrategy());
```

Comment:
You must do this before you call setProperties on the endpoint, as otherwise you override any custom endpoint configured heater filter strategy with the one from the component